include/envoy/network/socket_interface.h

```diff
@@ -49,13 +49,17 @@ class SocketInterface {
 using SocketInterfacePtr = std::unique_ptr<SocketInterface>;
 
 /**
- * Create IoHandle for given address
+ * Create IoHandle for given address.
  * @param type type of socket to be requested
  * @param addr address that is gleaned for address type, version and socket interface name
  * @return @ref Network::IoHandlePtr that wraps the underlying socket file descriptor
  */
 static inline IoHandlePtr ioHandleForAddr(Socket::Type type,
                                           const Address::InstanceConstSharedPtr addr) {
+  // Cannot create IoHandle for internal address yet.
+  if (addr->envoyInternalAddress()) {
```

Comment:
Unfortunately this looks like it's solid error handling but most callers of ioHandleForAddr will crash if it returns nullptr.

@mattklein123 @lambdai 's plan is basically creating this new listener type to be able to point a "cluster" at this new virtual address.  Any thoughts on how that might best be done without causing this cascading chain of not-fully-implemented things?

Reply:
I fixed this by providing the NullIoSocketHandleImpl to give it a socket handle allow upper layer to create a connection but will fail to send/recv.
https://github.com/envoyproxy/envoy/pull/12837/files#r483259684

Reply:
The ideal case is control plane never send unsupported addresses before envoy could support. 
1. Mark the cluster config invalid if it contains internal address. It's always safer to reject cluster configs at master.  
    I did this check in LDS config as well. That is easy since LDS is the only entrance to receive new listener config.
2. In case of escaped internal address which we fail to invalid, the NullIoSocketHandle could fail the per client connect.
    This NullSocketHandle could also support mingled addresses such as 
```
endpoints:
    - internal address: "not_yet_supported_listener_address"
    - socket address: "127.0.0.1:80"
```